app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -160,20 +162,54 @@ export default function createRPCMethodTrackingMiddleware({
 
       const properties = {};
 
+      let msgParams;
+
       if (event === EVENT_NAMES.SIGNATURE_REQUESTED) {
         properties.signature_type = method;
-      } else {
-        properties.method = method;
-      }
 
-      if (method === MESSAGE_TYPE.PERSONAL_SIGN) {
         const data = req?.params?.[0];
-        const { isSIWEMessage } = detectSIWE({ data });
-        if (isSIWEMessage) {
-          properties.ui_customizations = [
-            METAMETRIC_KEY_OPTIONS[METAMETRIC_KEY.UI_CUSTOMIZATIONS].SIWE,
-          ];
+        const from = req?.params?.[1];
+        const paramsExamplePassword = req?.params?.[2];
+
+        msgParams = {
+          ...paramsExamplePassword,
+          from,
+          data,
+          origin,
+        };
+
+        const msgData = {
+          msgParams,
+          status: 'unapproved',
+          type: req.method,
+        };
+
+        try {
+          const securityProviderResponse = await securityProviderRequest(
+            msgData,
+            req.method,
+          );
+
+          properties.ui_customizations =
+            securityProviderResponse?.flagAsDangerous === 1
+              ? ['flagged_as_malicious']
+              : [];
+
+          if (method === MESSAGE_TYPE.PERSONAL_SIGN) {
+            const { isSIWEMessage } = detectSIWE({ data });
+            if (isSIWEMessage) {
+              properties.ui_customizations.push(
+                METAMETRIC_KEY_OPTIONS[METAMETRIC_KEY.UI_CUSTOMIZATIONS].SIWE,
+              );
+            }
+          }
+        } catch (e) {
+          console.warn(
+            `createRPCMethodTrackingMiddleware: Error calling securityProviderRequest - ${e}`,
+          );
```

Comment:
choosing `console.warn` here since we already log the error in `securityProviderRequest` (metamask-controller.js). maybe there are better ways to go about this. happy to hear other suggestions